all-plugin-requirements.txt

```diff
@@ -3,7 +3,7 @@
 #
 
 # Provides fcm:// and spush://
-cryptography
+cryptography<3.4
```

Comment:
I'm not sure if this is a great idea to force a legacy version. I am definitely not a fan of the cryptography team switching to rust either, but during CI testing it's pretty easy to just force the <3.4 if needed... if at all.  I'm pretty sure test cases were passing previously with the setting as it was, no?

Reply:
I've just exercised the procedure within a Docker container, and it works because `pip` will just select the binary wheel for installation - there is no need for a Rust toolchain.
```
docker run --rm -it ubuntu:focal bash
apt update
apt install --yes python3-pip
pip install cryptography
```

```
Collecting cryptography
  Downloading cryptography-38.0.1-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)
     |████████████████████████████████| 4.1 MB 2.6 MB/s
Collecting cffi>=1.12
  Downloading cffi-1.15.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (442 kB)
     |████████████████████████████████| 442 kB 22.8 MB/s
Collecting pycparser
  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
     |████████████████████████████████| 118 kB 8.7 MB/s
Installing collected packages: pycparser, cffi, cryptography
Successfully installed cffi-1.15.1 cryptography-38.0.1 pycparser-2.21
```

I will probe if it was only on PyPy, where this croaked. In this case, and if you agree, I will just adjust the corresponding environment sections in `tox.ini` to specifically use `cryptography<3.4` there, similar to what you did before with `pip install --upgrade cryptography==3.1.1` due to OpenSSL vs. Python 3.5 linking woes.

Reply:
I've finally resolved it. Both d284300 and 4576702 were needed in order to tame PyPy into using `cryptography<3.4`. 

a) I think this is fine in general? All CPython versions will use the most recent version of `cryptography`, see [^1].
b) On this matter, I would like to question this section within `.travis.yml`, which seems redundant to me, because `tox` will apparently setup a dedicated virtualenv anyway [^2]. For improving it, let's handle it separately later.

[^1]: https://app.travis-ci.com/github/caronc/apprise/jobs/585078960#L363
[^2]: https://github.com/caronc/apprise/blob/b0bb560d055d2cb5f74aef07d40dd31cdee8b711/.travis.yml#L38-L43